addon/helpers/-base.js

```diff
@@ -0,0 +1,17 @@
+import Ember from 'ember';
+
+export default Ember.Helper.extend({
+  moment: Ember.inject.service(),
+
+  init() {
+    this._super(...arguments);
+    let service = this.get('moment');
+    service.addObserver('locale', this, this.recompute);
+  },
```

Comment:
I believe rather then the explicit observer (per instance), we can declaratively add via:
```js
localeDidChange: Ember.observer('moment.locale', function() {
  this.recompute();
})
```